### PR TITLE
Update device 'status' and 'isActive' fields

### LINKED ISSUE #380: "Expand" the deploy/Not Deployed State with more options
**Is your feature request related to a problem? Please describe.**
At the moment, the user managing the network is not able able to easily track the status of a newly created device in the production assembly line accordingly. The current implementation of netmanager does not offer options to monitor the hardware production process.

**Describe the solution you'd like**
Expanding the deploy/not deployed state with more options.
In assembly line =>  In-stock and ready  => Deployed => Un deployed -> Recalled for maintenance -> Decommissioned. 

**Describe alternatives you've considered**
- Utlliization of the "new" `status` field for the deployment status of the device instead of using the `isActive` field
- We need to review the UI colors we shall use for the different statuses. Currently we are use green for` deployed` and red for `undeployed`.

**Additional context**
This was picked up as a feature request from the recently concluded hardware demo.
API details:
We are using the update endpoint for devices where the updated field is `status`
There is already a device status field on the platform which we can take advantage of for this. Screenshot follows shortly:
Screenshot 2021-08-30 at 12 47 37

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
#### Summary of Changes (What does this PR do?)
- Adds more device statuses
- Closes #380 

**Note:** This PR depends on the ability to track site position inputs using the siteID on the device edit form

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* Checkout the [ft-addDeviceStatusOptions branch](https://github.com/airqo-platform/AirQo-frontend/tree/ft-addDeviceStatusOptions) and follow the instructions in the [README](https://github.com/airqo-platform/AirQo-frontend/blob/ft-addDeviceStatusOptions/netmanager/README.md) to run the application locally.
* Go to the [Device Registry section](http://localhost:5000/registry)
* You should see a variety of statuses on various devices.
* You can also choose to edit and add another deployment status

#### What are the relevant tickets?
- [PLAT-1010](https://airqoteam.atlassian.net/browse/PLAT-1010?atlOrigin=eyJpIjoiNjYzYWNmNzgwNmI1NDI0NjlhNWFhODZjMDJkNGVjZWMiLCJwIjoiaiJ9)
- #380

#### Screenshots (optional)
image

### CODE REVIEW

**Comment by 123MwanjeMike** [comment]
This PR depends on [this api PR](https://github.com/airqo-platform/AirQo-api/pull/746)

**Review by Baalmart — COMMENTED** [review]
Thanks @123MwanjeMike , I suggest that we use this same PR to address SOME or ALL of the feedback we have recently got from hardware team about the EDIT/DEPLOY page.

This included:

1. remove the latitude and longitude fields from all the forms since we can get them from the Site details.
2. Deal with the empty space "option" of the drop down menu for the new device status implementation.
3. Arrange the form in such a way that only few important items are shown during deployment or edit and others are only seen when one clicks to "view more". Or something along those lines. :) 

What are your thoughts? cc: @ja-odur

**Comment by 123MwanjeMike** [comment]
> Thanks @123MwanjeMike , I suggest that we use this same PR to address SOME or ALL of the feedback we have recently got from hardware team about the EDIT/DEPLOY page.
> 
> This included:
> 
>     1. remove the latitude and longitude fields from all the forms since we can get them from the Site details.
> 
>     2. Deal with the empty space "option" of the drop down menu for the new device status implementation.
> 
>     3. Arrange the form in such a way that only few important items are shown during deployment or edit and others are only seen when one clicks to "view more". Or something along those lines. :)
> 
> 
> What are your thoughts? cc: @ja-odur

Thanks @Baalmart. Let me incorporate on these changes asap

**Comment by Baalmart** [comment]
> > Thanks @123MwanjeMike , I suggest that we use this same PR to address SOME or ALL of the feedback we have recently got from hardware team about the EDIT/DEPLOY page.
> > This included:
> > ```
> > 1. remove the latitude and longitude fields from all the forms since we can get them from the Site details.
> > 
> > 2. Deal with the empty space "option" of the drop down menu for the new device status implementation.
> > 
> > 3. Arrange the form in such a way that only few important items are shown during deployment or edit and others are only seen when one clicks to "view more". Or something along those lines. :)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > What are your thoughts? cc: @ja-odur
> 
> Thanks @Baalmart. Let me incorporate on these changes asap

Awesome! Thanks @123MwanjeMike 
As for the worry of:
"if we remove the lat and long fields from the field, what coordinates shall we use for the device management map? Since one site can have more than one device, wont it create an overlap?"
**SUGGESTION:**
We could explore visualisation options where we can be able to distinguish devices at the same point in a map.

File: netmanager/src/views/components/DataDisplay/DeviceView/DeviceEdit.js [hunk 1]
```diff
@@ -256,6 +226,28 @@ const EditDeviceForm = ({ deviceData, siteOptions }) => {
...
+              onChange={handleSelectFieldChange("status")}
+              SelectProps={{
+                native: true,
+                style: { width: "100%", height: "50px" },
+              }}
+              variant="outlined"
+              error={!!errors.status}
+              helperText={errors.status}
+            >
+              <option value="" />
```
Comment (123MwanjeMike) [inline netmanager/src/views/components/DataDisplay/DeviceView/DeviceEdit.js:245]: @Baalmart, I think this empty space option was intentionally put in order to be the default selection before a user picks an option just like with "Internet Service Provider" and "Primary Device in Location" fields. @ja-odur could confirm this.

Comment (Baalmart) [reply netmanager/src/views/components/DataDisplay/DeviceView/DeviceEdit.js:245]: Alright @123MwanjeMike , sounds reasonable. Could we perhaps add some text like `None` or something? From the LIVE review by the hardware team, the empty space looked confusing. 
In summary, please feel free to use your creativity in eradicating that "confusion" we witnessed together...

Comment (123MwanjeMike) [reply netmanager/src/views/components/DataDisplay/DeviceView/DeviceEdit.js:245]: Sure thing, thanks for the go ahead @Baalmart

File: netmanager/src/views/components/DataDisplay/DeviceView/DeviceEdit.js [hunk 2]
```diff
@@ -33,6 +33,12 @@ const EDIT_OMITTED_KEYS = [
   "pictures",
 ];
 
+const AccordionDetails = styled(MuiAccordionDetails)(({ theme }) => ({
+    position: 'relative',
```
Comment (123MwanjeMike) [inline netmanager/src/views/components/DataDisplay/DeviceView/DeviceEdit.js:37]: This is meant to reduce the height of the accordion as requested by the platform team
image

**Review by Baalmart — COMMENTED** [review]
Thanks @123MwanjeMike , just a few comments that need your review....

- Amazing work in the device EDIT section, much appreciated!
- This PR has some new conflicts that need to be resolved.
- Also, the update in the device depioyment sub section needs a corresponding backend change present in the this [backend PR](https://github.com/airqo-platform/AirQo-api/pull/690). So we could either just fast track the complete review of this PR or just raised a new one that address just that small concern of being able to deploy with just site ID and no coordinates.

**Comment by 123MwanjeMike** [comment]
> Thanks @123MwanjeMike , just a few comments that need your review....
> 
>     * Amazing work in the device EDIT section, much appreciated!
> 
>     * This PR has some new conflicts that need to be resolved.
> 
>     * Also, the update in the device depioyment sub section needs a corresponding backend change present in the this [backend PR](https://github.com/airqo-platform/AirQo-api/pull/690). So we could either just fast track the complete review of this PR or just raised a new one that address just that small concern of being able to deploy with just site ID and no coordinates.

Thanks @Baalmart 
I think we can add the necessary changes to this PR.
For the issue of the conflicts, they to be resolved from the terminal and that includes pushing to the staging branch for which I have to permissions to do. Perhaps we could have a sync to have them resolved

**Review by Baalmart — APPROVED** [review]
LGTM

**Review by Baalmart — CHANGES_REQUESTED** [review]
there are some file conflicts....

**Review by Baalmart — APPROVED** [review]
LGTM

**Pull request merged**